plugins/modules/zammad_ticket.py

```diff
@@ -0,0 +1,476 @@
+    type: str
+    returned: always
+    sample: "Ticket created successfully."
+'''
+
+from ansible.module_utils.basic import AnsibleModule
+import json
+import requests
+
+def make_request(method, zammad_url, api_user, api_secret, data, ticket_id = None, endpoint = None):
+    headers = {"Content-type": "application/json"}
+    url = f"{zammad_url}/api/v1/tickets/{ticket_id}" if ticket_id else f"{zammad_url}/api/v1/{endpoint or 'tickets/'}"
+    try:
+        response = requests.request(method, url, data=json.dumps(data), headers=headers, auth=(api_user, api_secret))
+        response.raise_for_status()
+        return response.json(), response.status_code
+    except requests.exceptions.RequestException as e:
+        raise ValueError(f"API request failed: {e}")
+
+def create_ticket(zammad_url, api_user, api_secret, owner, customer, title, group, subject, body, internal, ticket_state, priority):
+	article = {
+		"subject": subject,
+		"body": body,
+		"type": "note",
+		"internal": str(internal).lower()
+	}
+
+	data = {
+		"article": article,
+		**{key: value for key, value in {
+			"owner_id": owner,
+			"title": title,
+			"group": group,
+			"state": ticket_state,
+			"customer": customer,
+			"priority": priority
+		}.items() if value is not None}
+	}
+
+	print(data)
+
+	return make_request("POST", zammad_url, api_user, api_secret, data)
+
+def update_ticket(zammad_url, api_user, api_secret, ticket_id, owner, customer, title, group, subject, body, internal, ticket_state, priority):
+	article = {}
+
+	if body: 
+		article = {
+			"subject": subject,
+			"body": body,
+			"type": "note",
+			"internal": str(internal).lower()
+		}
+
+	data = {
+		"article": article,
+		**{key: value for key, value in {
+			"owner_id": owner,
+			"title": title,
+			"group": group,
+			"ticket_state": ticket_state,
+			"priority": priority
+		}.items() if value is not None}
+	}
+	return make_request("PUT", zammad_url, api_user, api_secret, data, ticket_id)
+
+def close_ticket(zammad_url, api_user, api_secret, ticket_id):
+	data = {"state": "closed"}
+	return make_request("PUT", zammad_url, api_user, api_secret, data, ticket_id)
+
+def get_ticket(zammad_url, api_user, api_secret, ticket_id):
+	return make_request("GET", zammad_url, api_user, api_secret, {}, ticket_id)
+
+def get_users(zammad_url, api_user, api_secret):
+	return make_request("GET", zammad_url, api_user, api_secret, {}, endpoint = "users")
+
+def get_customer_name(ticket_data, customers):
+	customer_id = ticket_data.get("customer_id")
+	for customer in customers:
+		if customer["id"] == customer_id:
+			return customer["firstname"] + " " + customer["lastname"]
+
+def get_owner_name(ticket_data, owners):
+	owner_id = ticket_data.get("owner_id")
+	for owner in owners:
+		if owner["id"] == owner_id:
+			return owner["firstname"] + " " + owner["lastname"]
+
+def get_owner_id(owner_name, owners):
+	if owner_name == None:
+		return
+	firstname, lastname = owner_name.split()
```

Comment:
Does `owner_name` always follow the Firstname Lastname schema? What happens with middle names?

firstname = owner_name.split()[0]
lastname = owner_name.split()[-1]

might be a better approach, I'm not sure...